### ISSUE TITLE
Update travis build to test PyQt5

### ISSUE THREAD

**Issue by nicoddemus** [pytest-dev/pytest-qt#35]
As suggested by @jdreaver, an alternative is trying to use conda for that:

* http://conda.pydata.org/docs/travis.html
* https://binstar.org/jdreaver/pyqt5

**Comment by nicoddemus** [comment]
Hi @jdreaver,

I tried using your `pyqt5` package, but the [build](https://travis-ci.org/pytest-dev/pytest-qt/jobs/57285535) aborts right after starting py.test:

```
============================= test session starts ==============================
platform linux -- Python 3.4.3 -- py-1.4.26 -- pytest-2.7.0
qt-api: pyqt5
rootdir: /home/travis/build/pytest-dev/pytest-qt, inifile: 
plugins: qt
collected 31 items 
pytestqt/_tests/test_basics.py /home/travis/build.sh: line 41:  2617 Aborted 
```

I then remembered that one has to set `QT_QPA_PLATFORM_PLUGIN_PATH`, so I [did that](https://travis-ci.org/pytest-dev/pytest-qt/jobs/57286309) but the results are the same. :cry: 

Do you have any tips here?

Cheers,

**Comment by nicoddemus** [comment]
Not sure if `QT_QPA_PLATFORM_PLUGIN_PATH` is really required, but one can add [pre/post link scripts](http://conda.pydata.org/docs/build.html#pre-post-link-unlink-scripts). Just noting this here so I can easily find the reference again if needed. :smile:

**Comment by jdreaver** [comment]
I experienced this same thing today on my Ubuntu 14.04 continuous integration machine...

I ssh'ed into my server to try and get info on the `Aborted` message, and I got an error related to `xcb` being missing. Ideally, you should be able to just `conda install qt5` and have it work without having to install external dependencies (in this case, `xcb`). In order to do so, `qt5` needs to be built in such a way that all of the dependencies are either bundled or are installed as requirements in conda. I need to figure out more details about this error this week.

I built that particular `qt5` binstar package on Ubuntu 14.04, but maybe Travis and my build machine are missing some common dependency. 

In summary, I don't think there is a quick fix; the conda recipe for `qt5` needs to be fixed! :smile: I want to fix it this week.

**Comment by nicoddemus** [comment]
OK, thanks for the detailed explanation!

Would you let me know once you have more news, please? :smile:

Cheers, :beers:

**Comment by jdreaver** [comment]
I installed `libgl1-mesa-dri`, but that didn't fix the problem. If you are curious, here is what I get when I try to run my application with `LIBGL_DEBUG=verbose`:

```
libGL: screen 0 does not appear to be DRI3 capable
libGL: OpenDriver: trying /usr/lib/x86_64-linux-gnu/dri/tls/swrast_dri.so
libGL: OpenDriver: trying /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
libGL: driver does not expose __driDriverGetExtensions_swrast(): /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so: undefined symbol: __driDriverGetExtensions_swrast
libGL: Can't open configuration file /root/.drirc: No such file or directory.
libGL: Can't open configuration file /root/.drirc: No such file or directory.
libGL error: failed to load driver: swrast
Error: couldn't get an RGB, Double-buffered visual
```

After a ton of googling, I think the problem is that the `swrast_dri.so` version is broken in Ubuntu 14.04. However, it is possible this could work in Travis, which uses Ubuntu 12.04. I recommend these steps:

1. Install `libgl1-mesa-dri` in the `install` section of `.travis.yml`.
2. Add `LIBGL_DEBUG=verbose` as an environment variable so we can see the exact error.

**Comment by nicoddemus** [comment]
Hey @jdreaver,

Thanks for the update.

I tried what you suggested, but it seems `libgl1-mesa-dri` is already installed in travis, you can see the log [here](https://travis-ci.org/pytest-dev/pytest-qt/jobs/57719304).

**Comment by jdreaver** [comment]
Looks like it didn't work...

Oh well, I'll keep trying ideas!

**Comment by nicoddemus** [comment]
Yep. :disappointed:

**Comment by jdreaver** [comment]
I think I have it! Try this: 

```
xvfb-run -s "-screen 0 640x480x24" coverage run --source=pytestqt setup.py test
```

Basically, you pass the screen size and set **24 bits per pixel** (the `x24` part at the end). The default for xvfb is 8 bits per pixel. Now, I don't know if Ubuntu 12.04 has `xvfb-run`, so we have to pass in the arguments differently. There is a [section in the Travis docs](http://docs.travis-ci.com/user/gui-and-headless-browsers/#Configuring-xvfb-screen-size-and-more) on this. This is my best guess:
```
before_install:
  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x24"
```

I just took there example and set it to 24 bits per pixel. Note that you remove the `export DISPLAY=:99.0` line, since that is set in the above command. You can probably leave the `sleep 3` line.

**Comment by The-Compiler** [comment]
@nicoddemus can this be closed now that we can test PyQt5 on Ubuntu Trusty on Travis?

**Comment by nicoddemus** [comment]
Oh yes, thanks for the reminder! :smile:

**Comment by jdreaver** [comment]
Oh wow, I didn't know Travis updated Ubuntu! That simplifies a lot of CI deploys :+1:

**Comment by The-Compiler** [comment]
It didn't officially - it's just that you get Ubuntu Trusty when you use `sudo: required` and `services: docker` - no need to actually *use* docker then :wink:

**Comment by jdreaver** [comment]
Ah, that's clever!
